### PR TITLE
Refactorings based on Steffan153's work

### PR DESCRIPTION
**Pull request by ingydotnet**
These are now the best known yamlscript solutions.

### CODE REVIEW

**Comment by ingydotnet** [comment]
Requires 0.1.81 before merging.

File: exercises/practice/bank-account/.meta/bank-account.ys [hunk 1]
```diff
@@ -1,50 +1,28 @@
...
-  when amount > balance:
-    die: 'amount must be less than balance'
-  swap! account -: amount
-
-defn- get-amount(op):
-  amount =: op.amount
-  if amount < 0:
-    die: 'amount must be greater than 0'
-    else: amount
+  reduce _ closed operations:
```
Comment (vendethiel) [inline exercises/practice/bank-account/.meta/bank-account.ys:9]: I think this is a net negative. This loses the spirit of the exercise. The tests need to adjusted.

File: exercises/practice/hamming/.meta/hamming.ys [hunk 2]
```diff
@@ -3,5 +3,4 @@
 defn distance(strand1 strand2):
   strand1.# == strand2.# ||:
     die: 'strands must be of equal length'
-
-  len: map(ne strand1 strand2).filter(a)
```
Comment (vendethiel) [inline exercises/practice/hamming/.meta/hamming.ys:7]: this begs for a count occurences, not this hack

File: exercises/practice/list-ops/.meta/list-ops.ys [hunk 3]
```diff
@@ -19,10 +19,4 @@ defn foldl(list initial function):
...
 
 defn foldr(list initial function):
-  if len(list) > 0:
-    function _ list:first:
-      foldr: rest(list) initial function
-    else: initial
-
-defn reverse(list):
-  core/reverse: list
+  list:reverse.reduce(function initial)
```
Comment (vendethiel) [inline exercises/practice/list-ops/.meta/list-ops.ys:22]: this is also not a good example of reimplementing the language in the language, though it already wasn't...

File: exercises/practice/luhn/.meta/luhn.ys [hunk 4]
```diff
@@ -1,18 +1,6 @@
...
-      .conj(0).partition(2)
-      .map(munge):flat:sum
-      .mod(10).eq(0)
-
-defn munge([a b]):
-  b *=: 2
-  vector a:
-    if b > 9: (b - 9) b
+  digits =: value.replace(' ').map(\(_ - \\0)):reverse
+  digits.#.gt(1) &&: digits.every?(\(0 <= _ <= 9)) &&
```
Comment (vendethiel) [inline exercises/practice/luhn/.meta/luhn.ys:5]: fair for now, but also a bit of hack because of no precedence

**Review by vendethiel — APPROVED** [review]

File: exercises/practice/hamming/.meta/hamming.ys (hunk 2 above)
Comment (ingydotnet) [reply exercises/practice/hamming/.meta/hamming.ys:7]: Fixed

**Pull request merged**